### PR TITLE
Add parameter transparency to material appearance - issue 4252

### PR DESCRIPTION
**Pull request by amichel0205**
Add parameter transparency to material appearance.
Update UT test_02_create_material in test_03_Materials.py

### CODE REVIEW

File: pyaedt/modules/Material.py [hunk 1]
```diff
@@ -1353,46 +1356,60 @@ def __init__(self, materiallib, name, props=None, material_update=True):
 
     @property
     def material_appearance(self):
-        """Material Appearance specified as an RGB list.
+        """Material Appearance specified as a list where first 3 items are rgb color and fourth one is transparency.
```
Comment (MaxJPRey) [inline pyaedt/modules/Material.py:1359]:
```suggestion
        """Material appearance specified as a list where first 3 items are rgb color and fourth one is transparency.
```

File: pyaedt/modules/Material.py [hunk 2]
```diff
@@ -1353,46 +1356,60 @@ def __init__(self, materiallib, name, props=None, material_update=True):
...
-        """Material Appearance specified as an RGB list.
+        """Material appearance specified as a list where first 3 items are rgb color and fourth one is transparency.
 
         Returns
         -------
         list
-            Color of the material in RGB.  Values are in the range ``[0, 255]``.
+            Color of the material in RGB and transparency.
+            Color Values are in the range ``[0, 255]``.
+            Transparency is float in the range ``[0,1]``
```
Comment (PipKat) [inline pyaedt/modules/Material.py:1366]:
```suggestion
            Color values are in the range ``[0, 255]``.
            Transparency is a float in the range ``[0,1]``.
```

File: pyaedt/modules/Material.py [hunk 3]
```diff
@@ -1353,46 +1356,60 @@ def __init__(self, materiallib, name, props=None, material_update=True):
...
 
         Examples
         --------
-        Create a new material with color ``[0, 153, 153]`` (darker cyan).
+        Create a new material with color ``[0, 153, 153]`` (darker cyan) and transparency 0.5.
```
Comment (PipKat) [inline pyaedt/modules/Material.py:1370]:
```suggestion
        Create a material with color ``[0, 153, 153]`` (darker cyan) and transparency ``0.5``.
```

File: pyaedt/modules/Material.py [hunk 4]
```diff
@@ -1353,46 +1356,60 @@ def __init__(self, materiallib, name, props=None, material_update=True):
...
-        value_int = []
-        for rgb_item in rgb:
-            rgb_int = int(rgb_item)
-            if rgb_int < 0 or rgb_int > 255:
-                raise ValueError("RGB value must be between 0 and 255")
-            value_int.append(rgb_int)
-        self._material_appearance = value_int
+    def material_appearance(self, appearance_props):
+        if not isinstance(appearance_props, (list, tuple)):
+            raise TypeError("`material_appearance` must be a list or tuple")
```
Comment (PipKat) [inline pyaedt/modules/Material.py:1383]:
```suggestion
            raise TypeError("`material_appearance` must be a list or tuple.")
```
Message should be a complete sentence with concluding period.

File: pyaedt/modules/Material.py [hunk 5]
```diff
@@ -1353,46 +1356,60 @@ def __init__(self, materiallib, name, props=None, material_update=True):
...
+        if len(appearance_props) != 3 and len(appearance_props) != 4:
+            raise ValueError("`material_appearance` must be four items (R, G, B, transparency)")
```
Comment (PipKat) [inline pyaedt/modules/Material.py:1385]:
```suggestion
            raise ValueError("`material_appearance` must be four items (R, G, B, transparency).")
```

File: pyaedt/modules/Material.py [hunk 6]
```diff
@@ -1353,46 +1356,60 @@ def __init__(self, materiallib, name, props=None, material_update=True):
...
+        elif len(appearance_props) == 3:
+            transparency_value = self.material_appearance[3]
+            msg = "Only RGB specified, transparency is set to " + str(transparency_value)
```
Comment (PipKat) [inline pyaedt/modules/Material.py:1388]:
```suggestion
            msg = "Only RGB specified. Transparency is set to " + str(transparency_value)
```
Not sure where to place concluding period in this message string. Please add it.

File: pyaedt/modules/Material.py [hunk 7]
```diff
@@ -1353,46 +1356,60 @@ def __init__(self, materiallib, name, props=None, material_update=True):
...
+        for i in range(len(appearance_props)):
+            if i < 3:
+                rgb_int = int(appearance_props[i])
+                if rgb_int < 0 or rgb_int > 255:
+                    raise ValueError("RGB value must be between 0 and 255")
+                value.append(rgb_int)
+            else:
+                transparency = float(appearance_props[i])
+                if transparency < 0 or transparency > 1:
+                    raise ValueError("Transparency value must be between 0 and 1")
```
Comment (PipKat) [inline pyaedt/modules/Material.py:1401]:
```suggestion
                    raise ValueError("Transparency value must be between 0 and 1".)
```

**Review by PipKat — DISMISSED** [review]
Minor doc suggestions provided.

**Comment by amichel0205** [comment]
@PipKat 

Thanks for your help on making documentation cleaner.

**Pull request merged**